### ISSUE TITLE
Curly quotes in search queries make bad things happen

### ISSUE THREAD

**Issue by reefdog** [documentcloud/documentcloud#376]
Consider these four links, noting that `%E2%80%9C` is `“` (an opening curly quote) and `%E2%80%9D` is `”` (a closing curly quote):

```
https://www.documentcloud.org/search/Source:%20%E2%80%9CInternal%20documents%20from%20Mossack%20Fonseca%20(Panama%20Papers)%E2%80%9D
https://www.documentcloud.org/search/Source:%20%22Internal%20documents%20from%20Mossack%20Fonseca%20(Panama%20Papers)%22
https://www.documentcloud.org/search/Source:%20%E2%80%9CInternal%20documents%20from%20Mossack%20Fonseca%20(Panama%20Papers)%22
https://www.documentcloud.org/search/Source:%20%22Internal%20documents%20from%20Mossack%20Fonseca%20(Panama%20Papers)%E2%80%9D
```

Depending on which we use, you get one of three different experiences:

Style                                                                                                                                                             | Example | Docs                   | Searchbox contents
------------------------------------------------------------------------------------------------------------------------------------------------------------------|---------|------------------------|----------------------------------------------------------------------------
[Curly](https://www.documentcloud.org/public/search/Source:%20%E2%80%9CInternal%20documents%20from%20Mossack%20Fonseca%20(Panama%20Papers)%E2%80%9D)              | `“”`    | :white_check_mark: 151 | :x: Source: `“Internal` Text: `documents from Mossack Fonseca (Panama Papers)"`
[Straight](https://www.documentcloud.org/public/search/Source:%20%22Internal%20documents%20from%20Mossack%20Fonseca%20(Panama%20Papers)%22)                       | `""`    | :white_check_mark: 151 | :white_check_mark: Source: `Internal documents from Mossack Fonseca (Panama Papers)`
[Mismatch, curly first](https://www.documentcloud.org/search/Source:%20%E2%80%9CInternal%20documents%20from%20Mossack%20Fonseca%20(Panama%20Papers)%22)           | `“"`    | :white_check_mark: 151 | :x: Source: `“Internal` Text: `documents from Mossack Fonseca (Panama Papers)"`
[Mismatch, straight first](https://www.documentcloud.org/public/search/Source:%20%22Internal%20documents%20from%20Mossack%20Fonseca%20(Panama%20Papers)%E2%80%9D) | `"”`    | :x: 8                  | :x: Text: `Source: "Internal documents from Mossack Fonseca (Panama Papers)”`

So basically, Rails/Solr are handling paired quotes and curly-first mismatches okay, but barfing on straight-first mismatches. VisualSearch is barfing on any presence of curly quotes.

**Comment by reefdog** [comment]
Considering just having [`dc.controllers.Searcher#search()`](https://github.com/documentcloud/documentcloud/blob/a8b0b247d2239214161e21736974ffdb09403144/public/javascripts/app/searcher.js#L134) straighten out the curlies.

**Comment by reefdog** [comment]
Nah, never mind. Not so easy. By that point VS has already misunderstood the tokens. Here are perhaps better examples:

- Curly pair: https://www.documentcloud.org/public/search/Title:%20%E2%80%9CDithering%E2%80%9D
- Straight pair: https://www.documentcloud.org/public/search/Title:%20%22Dithering%22
- Mismatch, curly first: https://www.documentcloud.org/public/search/Title:%20%E2%80%9CDithering%22
- Mismatch, straight first: https://www.documentcloud.org/public/search/Title:%20%22Dithering%E2%80%9D

If you pop open the web console and enter `dc.app.visualSearch.searchQuery.models[0].attributes` you can see how it tokenized `Title`.
